### PR TITLE
delete unnecessary switch case judgement

### PR DESCRIPTION
**Pull request by flashJd**
private static RowData convert(Types.StructType struct, Record record) invoke private static Object convert(Type type, Object object), the latter has already use a switch case to handle STRUCT,LIST,MAP，so I just remove the unnecessary judgement  in the caller method.

### CODE REVIEW

File: flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/RowDataConverter.java [hunk 1]
```diff
@@ -62,22 +62,8 @@ private static RowData convert(Types.StructType struct, Record record) {
     List<Types.NestedField> fields = struct.fields();
     for (int i = 0; i < fields.size(); i += 1) {
       Types.NestedField field = fields.get(i);
-
```
Comment (rdblue) [inline flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/RowDataConverter.java:65]: Please revert this unnecessary whitespace change.

Comment (flashJd) [reply flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/RowDataConverter.java:65]: I have reverted the unnecessary whitespace change.

Comment (flashJd) [reply flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/RowDataConverter.java:65]: @rdblue anything do I need to modify?

**Comment by rdblue** [comment]
Thanks, @flashJd!
